File: API/talkx2api/talkx2api.py
from flask import Flask, request, jsonify
import uuid
import requests
import random
import time

app = Flask(__name__)

def generate_user_agent():
    os_list = [
        'Windows NT 10.0; Win64; x64',
        'Macintosh; Intel Mac OS X 10_15_7',
        'X11; Linux x86_64'
    ]

    browser_list = [
        ('Chrome', 'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/{}.{}.{}.{} Safari/537.36'),
        ('Firefox', 'Gecko/20100101 Firefox/{}.{}.{}'),
        ('Safari', 'AppleWebKit/605.1.15 (KHTML, like Gecko) Version/{}.{}.{} Safari/605.1.15'),
        ('Edge', 'Edge/{}.{}.{}.{}')
    ]

    os_choice = random.choice(os_list)
    browser_name, browser_format = random.choice(browser_list)

    major_version = random.randint(90, 122)
    minor_version = random.randint(0, 9)
    build_version = random.randint(0, 99)
    patch_version = random.randint(0, 99)

    if browser_name == 'Chrome':
        user_agent = f'Mozilla/5.0 ({os_choice}) {browser_format.format(major_version, minor_version, build_version, patch_version)}'
    elif browser_name == 'Firefox':
        user_agent = f'Mozilla/5.0 ({os_choice}; rv:{major_version}.0) {browser_format.format(major_version, minor_version, build_version)}'
    elif browser_name == 'Safari':
        user_agent = f'Mozilla/5.0 ({os_choice}) {browser_format.format(major_version, minor_version, build_version)}'
    else:
        user_agent = f'Mozilla/5.0 ({os_choice}) {browser_format.format(major_version, minor_version, build_version, patch_version)}'
    return user_agent

def get_header():
    headers = {
        'accept': '*/*',
        'accept-language': 'zh-CN,zh;q=0.9',
        'content-type': 'application/json;charset=utf-8',
        'origin': 'https://web.talkx.cn',
        'referer': 'https://web.talkx.cn/',
        'sec-ch-ua': '"Google Chrome";v="123", "Not:A-Brand";v="8", "Chromium";v="123"',
        'sec-ch-ua-mobile': '?0',
        'sec-ch-ua-platform': '"Windows"',
        'sec-fetch-dest': 'empty',
        'sec-fetch-mode': 'cors',
        'sec-fetch-site': 'same-site',
        'talkx-token': '',  # 在这里填入有效的 token
        'Accept-Charset': 'utf-8',
        'user-agent': generate_user_agent(),
    }
    return headers

def forward_request_to_talkx(messages):
    json_data = {
        'roleType': '0',
        'productId': 0,
        'sessionId': str(uuid.uuid4()),
        'messages': messages,
    }

    response = requests.post('https://api.talkx.cn/gpt/chat', headers=get_header(), json=json_data)
    response.encoding = 'utf-8'

    content_type = response.headers.get('Content-Type', '')

    if response.status_code == 200:
        if 'application/json' in content_type:
            return response.json()
        else:
            return {"choices": [{"message": {"role": "assistant", "content": response.text}}]}
    else:
        return {"error": "Failed to fetch response from talkx API", "status_code": response.status_code, "content": response.text}

@app.route("/v1/chat/completions", methods=["POST"])
def chat_completions():
    openai_request = request.json
    messages = [{"role": message["role"], "content": message["content"]} for message in openai_request.get("messages", [])]

    talkx_response = forward_request_to_talkx(messages)

    openai_response = {
        "id": str(uuid.uuid4()),
        "object": "chat.completion",
        "created": int(time.time()),
        "model": "gpt-3.5-turbo",
        "choices": talkx_response.get("choices", []),
        "usage": {
            "prompt_tokens": len(openai_request.get("messages", [])),
            "completion_tokens": len(talkx_response.get("choices", [])),
            "total_tokens": len(openai_request.get("messages", [])) + len(talkx_response.get("choices", []))
        }
    }

    return jsonify(openai_response)

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=5000)

File: API/talkx2api/talkx2api.js
addEventListener('fetch', event => {
  event.respondWith(handleRequest(event.request))
})

async function handleRequest(request) {
  try {
    // 解析请求体
    const requestData = await request.json()
    const messages = requestData.messages || []

    // 构建要发送到 talkx API 的请求体
    const talkxPayload = {
      roleType: '0',
      productId: 0,
      sessionId: generateUUID(),
      messages: messages.map(msg => ({
        role: msg.role,
        content: msg.content
      }))
    }

    // 向 talkx API 发送请求
    const talkxResponse = await fetch('https://api.talkx.cn/gpt/chat', {
      method: 'POST',
      headers: {
        'accept': '*/*',
        'accept-language': 'zh-CN,zh;q=0.9',
        'content-type': 'application/json;charset=utf-8',
        'origin': 'https://web.talkx.cn',
        'referer': 'https://web.talkx.cn/',
        'sec-ch-ua': '"Google Chrome";v="123", "Not:A-Brand";v="8", "Chromium";v="123"',
        'sec-ch-ua-mobile': '?0',
        'sec-ch-ua-platform': '"Windows"',
        'sec-fetch-dest': 'empty',
        'sec-fetch-mode': 'cors',
        'sec-fetch-site': 'same-site',
        'talkx-token': '',
        'user-agent': generateUserAgent(),
      },
      body: JSON.stringify(talkxPayload)
    })

    // 处理 talkx API 的响应
    const contentType = talkxResponse.headers.get('Content-Type') || ''
    let talkxData

    if (contentType.includes('application/json')) {
      talkxData = await talkxResponse.json()
    } else {
      const textResponse = await talkxResponse.text()
      talkxData = {
        choices: [
          {
            message: {
              role: 'assistant',
              content: textResponse
            }
          }
        ]
      }
    }

    // 构建 OpenAI 标准格式的响应
    const openaiResponse = {
      id: generateUUID(),
      object: 'chat.completion',
      created: Math.floor(Date.now() / 1000),
      model: 'gpt-3.5-turbo',
      choices: talkxData.choices || [],
      usage: {
        prompt_tokens: messages.length,
        completion_tokens: talkxData.choices.length,
        total_tokens: messages.length + talkxData.choices.length
      }
    }

    // 返回响应
    return new Response(JSON.stringify(openaiResponse), {
      headers: { 'Content-Type': 'application/json' }
    })

  } catch (error) {
    return new Response(JSON.stringify({ error: 'Internal Server Error' }), {
      status: 500,
      headers: { 'Content-Type': 'application/json' }
    })
  }
}

// 生成随机 UUID
function generateUUID() {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
    const r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8)
    return v.toString(16)
  })
}

// 生成随机 User-Agent
function generateUserAgent() {
  const osList = [
    'Windows NT 10.0; Win64; x64',
    'Macintosh; Intel Mac OS X 10_15_7',
    'X11; Linux x86_64'
  ]

  const browserList = [
    `Chrome/${getRandomInt(90, 122)}.${getRandomInt(0, 9)}.${getRandomInt(0, 99)}.${getRandomInt(0, 99)}`,
    `Firefox/${getRandomInt(90, 122)}.${getRandomInt(0, 9)}.${getRandomInt(0, 99)}`,
    `Safari/605.1.15`
  ]

  const os = osList[Math.floor(Math.random() * osList.length)]
  const browser = browserList[Math.floor(Math.random() * browserList.length)]

  return `Mozilla/5.0 (${os}) AppleWebKit/537.36 (KHTML, like Gecko) ${browser} Safari/537.36`
}

// 生成随机整数
function getRandomInt(min, max) {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min + 1)) + min
}
